### PR TITLE
#10-Implement "Hero Banner" Block 

### LINKED ISSUE #10: Implement hero
We should support both video background and composing images:

Composing images:
Original: https://www.esri.com/en-us/capabilities/imagery-remote-sensing/capabilities/management#contact
New: https://main--esri--aemsites.hlx.page/en-us/capabilities/imagery-remote-sensing/capabilities/management

image

Video background:
Original page: https://www.esri.com/en-us/about/about-esri/overview
New: https://main--esri--aemsites.hlx.page/en-us/about/about-esri/overview

image

- [x] Implement hero with composing images
- [x] Implement background video autoplay
- [x] #11

### PR DESCRIPTION
**Pull request by jsavella**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Partial fix for #10
- Implement "Hero Banner " block styling (xlarge, large, medium, small / Mobile)
- Leveraged calcite variables, spacing, font size, colors
- Pending: Calcite Buttons

Test URLs:
- Original: https://www.esri.com/en-us/about/about-esri/overview
- Before: https://main--esri--aemsites.hlx.live/en-us/about/about-esri/overview
- After: https://hero-banner--esri--aemsites.hlx.live/en-us/about/about-esri/overview

### CODE REVIEW

File: blocks/hero/hero.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Variant Sizes: xlarge, large (Default), medium, small
+ */
+export default function decorate() {
```
Comment (alexcarol) [inline blocks/hero/hero.js:4]: I see a lot of changes to the markup here, some guidance on best practices:
1. Use the "block" that is passed to decorate ( passed here: https://github.com/aemsites/esri/blob/105c78c8f0965c16248dccc890172477bfa40554/scripts/aem.js#L632)
2. Do not change the main structure (.section > .<block_name>-wrapper > .<block name>)
3. Try to minimize changes to the markup: before making a change to the markup (adding a class/changing the default structure) make sure that it brings value.

**Comment by alexcarol** [comment]
What should be fixed in this PR: 
- Performance: Fix CLS issues
- Best practices (mentioned in an inline comment on the hero)

Other stuff that's pending:
- Fix autoplay (https://gist.github.com/amol-anand/f797e6ab9513898e132987dabcaca262 from https://www.aem.live/developer/block-collection#block-party)
- Fixing text style and placement


Note: I've updated the task description from "Fix" to "Partial fix for" to prevent autoclosing the issue when this is merged

File: blocks/hero/hero.js (hunk 1 above)
Comment (jsavella) [reply blocks/hero/hero.js]: - Used "block" param.  
- Minimize markup changes. 
- Will need script to dynamically build a <video> element with <source src='video url'>. Similar to dynamically built <picture> element with queries.

**Review by tim-white-esri — COMMENTED** [review]
Simplify the CSS down. No reason to have `.hero.block` That just increased specificity. `.bloc` should be all you need. 

Same with multiple selectors. Keep it as simple as possible.

File: blocks/hero/hero.css [hunk 2]
```diff
@@ -1,34 +1,155 @@
 main .hero-container > div {
```
Comment (tim-white-esri) [inline blocks/hero/hero.css:1]: No need for `main`

File: blocks/hero/hero.css [hunk 3]
```diff
@@ -1,34 +1,155 @@
...
+  display: flex;
+  inline-size: 100%;
+  block-size: 100%;
+  align-items: center;
+  color: var(--calcite-ui-text-1);
+}
+
+.hero.block > div > div {
+  display: block;
+}
```
Comment (tim-white-esri) [inline blocks/hero/hero.css:39]: div is block by default.

File: blocks/hero/hero.css (hunk 2 above)
Comment (jsavella) [reply blocks/hero/hero.css]: removed

File: blocks/hero/hero.css (hunk 3 above)
Comment (jsavella) [reply blocks/hero/hero.css]: removed

**Pull request merged**